frontend/src/pages/LineageView.tsx

```diff
@@ -115,29 +115,90 @@ class LineageView extends React.Component<LineageViewProps, LineageViewState> {
-    // this.setState({
-    //   rows: this.getRowsFromArtifacts(request),
-    // });
-    return '';
+  private buildExecutionCards(executions: Execution[]): CardDetails[] {
+    return [
+      {
+        title: 'Execution',
+        elements: executions.map((execution) => ({
+            resource: execution,
+            prev: true,
+            next: true
+          })
+        )
+      }
+    ];
+  }
+
+  private async loadData(id: number): Promise<string> {
+    // TODO: Optimize async calls with Promise.all()
+    const getEventsByArtifactIDsRequest = new GetEventsByArtifactIDsRequest();
+    getEventsByArtifactIDsRequest.addArtifactIds(id);
+
+    const getEventsByArtifactIDsResponse =
+      await this.metadataStoreService.getEventsByArtifactIDs(getEventsByArtifactIDsRequest);
+
+    const events = getEventsByArtifactIDsResponse.getEventsList();
+
+    const outputExecutionIds =
+      events.filter(isOutputEvent).map((event) => (event.getExecutionId()));
+    const inputExecutionIds =
+      events.filter(isInputEvent).map((event) => (event.getExecutionId()));
+
+    const outputExecutions = await this.getExecutions(outputExecutionIds);
+    const inputExecutions = await this.getExecutions(inputExecutionIds);
+
+    // Build the list of input execution events.
+    // The list of input executions is the list of events in which the target
+    // artifact was an output artifact.
+    const getInputExecutionEvents = new GetEventsByExecutionIDsRequest();
+    getInputExecutionEvents.setExecutionIdsList(outputExecutionIds);
+    const getInputExecutionEventsResponse =
+      await this.metadataStoreService.getEventsByExecutionIDs(getInputExecutionEvents);
+
+    // Build the list of input artifacts for the input execution
+    const inputExecutionInputArtifactIds =
+      getInputExecutionEventsResponse
+        .getEventsList()
+        .filter(isInputEvent)
+        .map((event) => event.getArtifactId());
+    const inputArtifacts = await this.getArtifacts(inputExecutionInputArtifactIds);
+
+    // Build the list of output execution artifacts.
+    // The list of output executions is the list of events in which the target
+    // artifact was an input artifact.
+    const getOutputExecutionEventsRequest = new GetEventsByExecutionIDsRequest();
+    getOutputExecutionEventsRequest.setExecutionIdsList(inputExecutionIds);
+    const getOutputExecutionEventsResponse =
+      await this.metadataStoreService.getEventsByExecutionIDs(getOutputExecutionEventsRequest);
+
+    const outputExecutionOutputArtifactIds =
+      getOutputExecutionEventsResponse
+        .getEventsList()
+        .filter(isOutputEvent)
+        .map((event) => event.getArtifactId());
+    const outputArtifacts = await this.getArtifacts(outputExecutionOutputArtifactIds);
+
+    this.setState({
+      inputArtifacts: inputArtifacts,
+      inputExecutions: outputExecutions,
+      outputExecutions: inputExecutions,
+      outputArtifacts: outputArtifacts,
+    });
```

Comment:
Condense the entire object as so:
```suggestion
    this.setState({
        inputArtifacts, inputExecutions, outputExecutions, outputArtifacts
    });
```

Reply:
Done

Reply:
Done.